src/pages/Home/index.tsx

```diff
@@ -1,76 +1,80 @@
-import { useSignal, useSignalEffect } from "@preact/signals";
+import { useLocation } from "preact-iso";
+import { useSignalEffect } from "@preact/signals";
+import { peer } from "../../utils/peer";
+import {
+  createRoom,
+  joinRoom,
+  playerName,
+  roomName,
+} from "../../utils/session";
 import "./style.css";
-import { useRef } from "preact/hooks";
 
 export function Home() {
-  const buttonRef = useRef<HTMLButtonElement>();
-  const dialogRef = useRef<HTMLDialogElement>();
-  const dialogOpened = useSignal(false);
+  const { route } = useLocation();
```

Comment:
I felt like we should make `const { route } = useLocation();` a until function since other components are also using it.

Reply:
But this is a hook bonding to component scope, I do not think we can extract it.